### DISCUSSION TITLE
Activity with conditional properties

### DISCUSSION THREAD

**Discussion by programatix** [elsa-workflows/elsa-core#4366] [Q&A]
Hi,

Is it possible to add conditional properties. For example,

Input: Radio List
- -- Select One --
- Child by Name
- Child by Identifier

Input: Child Name
Input: Child Identifier

When the editor is open, the `Radio List` is `-- Select One --`, thus both the `Child Name` and `Child Identifier` are either hidden or disabled. When `Child by Name` is selected, then the `Child Name` is enable (or shown) and vice versa.

I have looked at https://v2.elsaworkflows.io/docs/extensibility/designer-plugins#custom-field-types and is thinking of creating a custom Radio List which will look for other Input and disable/hide or enable/unhide them. I'm not sure whether it's possible, and decide to ask first.

Any advice is greatly appreciated.

**Comment by sfmskywalker** [comment]
Hi @programatix,

Yes, it's possible to add conditional properties as you described. You would need to implement a custom field type to achieve this behavior.

For guidance, you can take a look at an existing example in the Elsa codebase: [send-http-request-plugin.tsx](https://github.com/elsa-workflows/elsa-core/blob/master/src/designer/elsa-workflows-studio/src/plugins/send-http-request-plugin.tsx). This should give you an idea of how to structure and implement your custom field type.
